### PR TITLE
Droth 743 municipality api

### PR DESCRIPTION
**Pull request by joanacoelho**

### CODE REVIEW

File: digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/MunicipalityApi.scala [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    }
+  }
+
+  def validateSideCodes(assets: Seq[NewLinearAsset]) : Unit = {
+    assets.map( _.sideCode )
+      .foreach( sc =>
+        if( !SideCode.values.map(_.value).contains(sc))
+          halt(UnprocessableEntity("Side code doesn't have a valid code."))
+      )
+  }
```
Comment (MAC-CGI) [inline digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/MunicipalityApi.scala]: "Unknown" sidecode value should be possible?

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/OnOffLinearAssetService.scala [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    //Expire the old asset
+    dao.updateExpiration(assetId, expired = true, username)
+
+    measures.map {
+      measure =>
+        Seq(Measures(oldAsset.startMeasure, measure.startMeasure), Measures(measure.endMeasure, oldAsset.endMeasure)).map {
+          m =>
+            if (m.startMeasure - m.endMeasure != 0)
+              createWithoutTransaction(oldAsset.typeId, oldAsset.linkId, valueToUpdate, sideCode.getOrElse(oldAsset.sideCode),
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/OnOffLinearAssetService.scala]: Maybe we should have some tolerance between "m.startMeasure - m.endMeasure"

**Comment by joanacoelho** [comment]
ok

**Comment by oskarmalmstrom** [comment]
ok

**Comment by marquesfm** [comment]
ok

**Pull request merged**